src/client/regionalClient.ts

```diff
@@ -1,35 +1,90 @@
 import { ServiceClient } from './client';
 
+const DEFAULT_TIMEOUT = 20000;
+
 export type RegionalResourceIteratee<T> = (each: {
   resource: T;
   region: string;
 }) => Promise<void> | void;
 
+interface RegionalServiceClientOptions {
+  logger: IntegrationLogger;
+  accessKeyId: string;
+  accessKeySecret: string;
+  apiVersion: string;
+  timeout?: number;
+}
+
+export interface RegionList {
+  regionId: string;
+  regionEndpoint: string;
+}
+
 export abstract class RegionalServiceClient extends ServiceClient {
-  protected getRegions: () => Promise<string[]>;
+  protected getRegions: () => Promise<RegionList[]>;
+  private regions: string[];
+  private regionalClients: AlibabaClient[];
+  private clientConfig;
+  private timeout: number | undefined;
 
-  constructor(options: { logger: IntegrationLogger }) {
+  constructor(options: RegionalServiceClientOptions) {
     super(options);
+
+    const { logger, timeout, ...config } = options;
+
+    this.clientConfig = config;
+    this.timeout = timeout;
+    this.regions = [];
+    this.regionalClients = [];
   }
 
-  protected async forEachRegion(cb: (region: string) => Promise<void>) {
-    const regions = await this.getRegions();
-    const errors: any[] = [];
-    for (const region of regions) {
-      try {
-        await cb(region);
-      } catch (err) {
-        err.region = region;
-        errors.push(err);
+  private async getRegionalClients() {
+    // I don't know if this 'optimization' will do any good
+    // If you need to restart whenever getRegions payload updates
+    // if (this.regionalClients.length > 0) return;
+
+    const regionList = await this.getRegions();
+
+    this.regionalClients = regionList.map(
+      (r) =>
+        new AlibabaClient({
+          ...this.clientConfig,
+          endpoint: 'https://' + r.regionEndpoint,
+          opts: {
+            timeout: this.timeout || DEFAULT_TIMEOUT,
+          },
+        }),
+    );
+    this.regions = regionList.map((r) => r.regionId);
+  }
+
+  protected async forEachRegion(
+    cb: (client: AlibabaClient, region: string) => Promise<void>,
+  ) {
+    await this.getRegionalClients();
+
+    const cbRoutines = this.regionalClients.map((client, i) =>
+      cb(client, this.regions[i]),
+    );
+    const results = await Promise.allSettled(cbRoutines);
+
+    let errorCount = 0;
+    const failingRegions: string[] = [];
+
+    results.forEach((res, i) => {
+      if (res.status === 'rejected') {
+        errorCount++;
+        failingRegions.push(this.regions[i]);
       }
-    }
+    });
 
-    if (errors.length > 0) {
-      const err = errors[0];
-      err.additionalErrors = errors.slice(1);
-      err.additionalErrorsLength = errors.length;
-      throw err;
+    if (errorCount > 0) {
+      this.logger.warn(`Service has ${errorCount} failing region(s)`);
+      this.logger.warn(
+        'Failing regional requests: ' + failingRegions.join(','),
+      );
     }
```

Comment:
Here's something that may be a bit easier on the eyes. Do with it what you wish :smile:

```
    const errors = results.filter((res) => res.status === 'rejected');
    const failingRegions = errors.map((_, i) => this.regions[i]);

    if (errors.length > 0) {
      this.logger.warn(`Service has ${errors.length} failing region(s)`);
      this.logger.warn(
        `Failing regional requests: ${failingRegions.join(',')}`,
      );
    }
```

Also one question: is there any scenario in which the regions could be in an unexpected order? (referring to the i index) If so, might be better to match using something like region id, if one is available.